projects/golang/math_big_fuzzer.go

```diff
@@ -0,0 +1,47 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+
+package mathfuzzer
+
+import "math/big"
+
+func FuzzCmp(data []byte) int {
+    if !isDivisibleBy(len(data), 2) {
+        return -1
+    }
+    i1 := new(big.Int)
+    i2 := new(big.Int)
+
+    half := len(data) / 2
+
+    halfOne := data[:half]
+    halfTwo := data[half:]
+
+    i1.SetBytes(halfOne)
```

Comment:
I realize I may have led you slightly astray here, so apologies for that. Using `SetBytes` could work, but it sort of depends what you want to test here. If you want FuzzCmp to accept something like []byte("123456") and have it turn into i1=123 and i2=456, then you'll want to do something more like the following:

```
x, y := new(big.Int), new(big.Int)
half := len(data)/2
if err := x.UnmarshalText(data[:half]); err != nil {
  return 0
}
if err := y.UnmarshalText(data[half:]); err != nil {
  return 0
}
x.Cmp(y)
return 1
``` 

Otherwise, what you have now will work, but it will convert bytes that aren't integer values into a `*big.Int` value, which might not be exactly what you want. I personally think using `UnmarshalText` is a bit cleaner.

Reply:
Why not do both?

Reply:
Sure, you can make two fuzz targets that parse the bytes in two different ways if you're looking to further expand coverage.

Reply:
Cool, I added another fuzzer with your suggestion.